src/wallet/api.rs

```diff
@@ -913,6 +926,37 @@ impl Wallet {
         Ok((privkey, public_key))
     }
 
+    /// Refreshes the UTXO cache using the wallet's internal spend info checks.
+    pub(crate) fn update_utxo_cache(&mut self, utxos: Vec<ListUnspentResultEntry>) {
+        // Process UTXOs to pair each with its spend info using the wallet's private methods.
+        let processed_utxos = utxos
+            .iter()
+            .filter_map(|utxo| {
+                // Try each check in order.
+                let mut spend_info = self.check_if_fidelity(utxo);
+                if spend_info.is_none() {
+                    spend_info = self.check_if_live_contract(utxo).unwrap();
+                }
+                if spend_info.is_none() {
+                    spend_info = self.check_descriptor_utxo_or_swap_coin(utxo).unwrap();
+                }
+                spend_info.map(|info| (utxo.clone(), info))
+            })
+            .collect::<Vec<(ListUnspentResultEntry, UTXOSpendInfo)>>();
+
+        // Clear the existing cache
+        self.store.utxo_cache.clear();
+
+        // Insert each processed UTXO into the cache using its OutPoint as key.
+        for (utxo, spend_info) in processed_utxos {
```

Comment:
This can be expensive for wallets having a lot of utxos. Not a big deal for now. 

Mark here a TODO: Implement smarter logic to add/remove entry from the utxo map. debug log the addition and deletions.

This will also allow us to track utxo movements in debug logs if required. This can be done as a followup enhancement PR. Great if you can also open up an issue tracking this.